test/util/removeUnnecessaryParens.test.js

```diff
@@ -22,7 +22,7 @@ describe('removeUnnecessaryParens', function () {
     ['(((-5)))', '-5'],
     ['((4+5)) + ((2^3))', '(4 + 5) + 2^3'],
     ['(2x^6 + -50 x^2) - (x^4)', '2x^6 - 50x^2 - x^4'],
-    ['(x+4) - (12 + x)', 'x + 4 - (12 + x)'],
+    ['(x+4) - (12 + x)', '(x + 4) - (12 + x)'],
```

Comment:
Actually, if I flattenOperands(node) in print, it heavily breaks tests (about 20 tests). I suppose simplify relied on print in some way. That is why I did't do it in first place.